src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1519,7 +1521,7 @@ ModifyQuerySupported(Query *queryTree)
 			}
 
 			if (commandType == CMD_INSERT && targetEntryPartitionColumn &&
-				!IsA(targetEntry->expr, Const))
+				ExpressionContainsParam((Node *) targetEntry->expr))
```

Comment:
Not quite, `EvaluateNodeIfReferencesFunction` doesn't always evaluate parameters. You're right that this could be supported now, and it could eventually replace our prepared statement hack, but I'd rather do that in a separate PR.